### PR TITLE
Upgrade ember 2.12+. Using mirage for testing. Adding shallow copying

### LINKED ISSUE #24: release a new version
Would be cool to have a new version with all the fixes :).

### PR DESCRIPTION
**Pull request by nyon**
First, thanks for this great package. I use it a lot in my administration interfaces.

However, I stumbled upon some issues and began to refactor and improve. The following changes were made:

- [x] Upgrading from ember 1.x to ember 2.12 (should resolve #24 )
- [x] Migrating code from ES5 to ES6
- [x] Minor refactoring of `Copyable` mixin
- [x] removing old code that based on `DS.FixtureAdapter` and replaced it with `ember-cli-mirage`
- [x] ignoring relations that could not be resolved (a must have for me)
- [x] adding shallow copying as a new method (`copy` and `shallowCopy`)

### CODE REVIEW

**Comment by milindalvares** [comment]
@lazybensch Pls review

**Comment by jfrux** [comment]
Definitely appears to be the most productive of the merges.
I vote for this one to override the current since the original repo owner is asking for new maintainers.

**Comment by jfrux** [comment]
Does this one happen to fix the inverse relationships issue?

**Review by jfrux — APPROVED** [review]
I fully support these changes.

**Comment by stevesims** [comment]
Big 👍  from me on this - would be great to see this merged

File: addon/mixins/copyable.js [hunk 1]
```diff
@@ -1,136 +1,181 @@
...
+    if(!copy) {
+      return;
+    }
+
+    for(let optionName in options) {
+      let option = options[optionName];
+      if(emberTypeOf(option) === 'object') {
+        this._applyOptions(option, get(copy, optionName));
+      } else {
+        set(copy, optionName, option);
```
Comment (stevesims) [inline addon/mixins/copyable.js:53]: I'm seeing a crash at this line of `Assertion Failed: You must pass an array of records to set a hasMany relationship` when attempting to make a copy of a complex record (with several relationships), where I had been attempting to skip copying a relationship by passing in an options object like `{ relationshipName: null }`.

Comment (nyon) [reply addon/mixins/copyable.js:53]: Could you fork my forked repo and provide a test? I'll have a look into it then.

**Comment by erkie** [comment]
Need any help here to get this merge?

**Comment by erkie** [comment]
I feel like a good way forward would be to merge this PR as is, and then we can create smaller PR's to fix any issues that arise. It's also easier to just point `package.json` to the master branch so we can start testing it. It's too good of a change on the whole to let die in PR state like this.

**Pull request closed without merging**